tools/lint/check_destructive_migration_marker.py

```
@@ -0,0 +1,146 @@
+    #    and reports clean is indistinguishable from a clean input.
+    if revision is None:
+        return [f"{path.name}: no readable module-level `revision` assignment; the floor cannot be applied"]
+
+    # 2. Undated revision -> fail, naming the CONVENTION rather than the floor.
+    #    A revision the floor cannot be compared against is a different defect
+    #    from a missing marker and deserves different text.
+    if not DATED_REVISION.match(revision) and revision not in PRE_CONVENTION:
+        return [
+            f"{path.name}: revision {revision!r} is not in this repository's "
+            f"YYYYMMDD_NNN[x] form, so it cannot be ordered against the floor. "
+            f"Alembic's default hex id sorts arbitrarily either side of it."
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)